src/bin/deploy.rs

```diff
@@ -161,38 +165,83 @@ enum GetDeploymentDataError {
 /// Evaluates the Nix in the given `repo` and return the processed Data from it
 async fn get_deployment_data(
     supports_flakes: bool,
-    repo: &str,
+    strict_eval: bool,
+    flake: &deploy::DeployFlake<'_>,
     extra_build_args: &[String],
 ) -> Result<deploy::data::Data, GetDeploymentDataError> {
-    info!("Evaluating flake in {}", repo);
+    info!("Evaluating flake in {}", flake.repo);
 
-    let mut c = match supports_flakes {
-        true => Command::new("nix"),
-        false => Command::new("nix-instantiate"),
+    let mut c = if supports_flakes {
+        Command::new("nix")
+    } else {
+        Command::new("nix-instantiate")
     };
 
-    let mut build_command = match supports_flakes {
-        true => {
-            c.arg("eval")
+    if supports_flakes {
+        c
+            .arg("eval")
             .arg("--json")
-            .arg(format!("{}#deploy", repo))
-        }
-        false => {
-            c
-                .arg("--strict")
-                .arg("--read-write-mode")
-                .arg("--json")
-                .arg("--eval")
-                .arg("-E")
-                .arg(format!("let r = import {}/.; in if builtins.isFunction r then (r {{}}).deploy else r.deploy", repo))
+            .arg(format!("{}#deploy", flake.repo));
+        match (&flake.node, strict_eval) {
```

Comment:
If there's a clean way to throw `flake.profile` into this match statement too, instead of having it as a child, then I would much prefer that.

Reply:
Refactored into a single `match`